src/Microsoft.NetCore.Analyzers/Core/Security/SetViewStateUserKey.cs

```diff
@@ -0,0 +1,101 @@
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.SetViewStateUserKeyDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var compilation = compilationStartAnalysisContext.Compilation;
+                var pageTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemWebUIPage);
+
+                if (pageTypeSymbol == null)
+                {
+                    return;
+                }
+
+                var eventArgsTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+
+                if (eventArgsTypeSymbol == null)
+                {
+                    return;
+                }
+
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
+                {
+                    var classSymbol = (INamedTypeSymbol)symbolAnalysisContext.Symbol;
+                    var baseClassSymbol = classSymbol.BaseType;
+
+                    if (baseClassSymbol.Equals(pageTypeSymbol))
+                    {
+                        var onInitMethodSymbol = classSymbol.GetMembers().OfType<IMethodSymbol>().FirstOrDefault(s => s.Name == "OnInit" &&
+                                                                                                        s.Parameters.Length == 1 &&
+                                                                                                        s.Parameters[0].Type.Equals(eventArgsTypeSymbol) &&
+                                                                                                        s.IsProtected());
```

Comment:
General question: does the parameter here have to `eventArgsTypeSymbol` or any custom event args type derived from `eventArgsTypeSymbol`?

Reply:
Believe the only overload is EventArgs: https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.page.oninit